### PR TITLE
Use openshift builder ImageStreamTag from other projects when internal registry is used.

### LINKED ISSUE #34243: Investigate if it makes sense to allow Openshift users to reuse ImageStream from other namespaces.
### Description

It was brought to my attention that users that need to use an Openshift ImageStream (e.g a builder image) from another namespace have difficulties as we currently have no way of expressing this need.

It appears that this is something one can do when defining the ImageStream manually, but not possible when using the `quarkus-openshift` extension.

A workaround is to use the full image (internal registry, repo and tag), but this may require RBAC recofniguration as in once case user reads streams, while in the other image stream tags.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by iocanel**
Resolves: #34243

### CODE REVIEW

**Review by Sgitario — CHANGES_REQUESTED** [review]
The changes look good to me. Though, this use case should be documented in the guides and also in the config properties (the `base jvm` and the `base native` properties).

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 1]
```diff
@@ -137,6 +137,12 @@ To trigger a container image build:
 
 The build that will be performed is a _s2i binary_ build.
 The input of the build is the jar that has been built locally and the output of the build is an `ImageStream` that is configured to automatically trigger a deployment.
+The base/builder image is specified using `baseJvmImage` and `baseNativeImage` for jvm and native mode respectively. An ImageStream for the image is automatically generated, unless these properties are used to reference an existing ImageStreamTag in the internal openshift registry. For example:
```
Comment (Sgitario) [inline docs/src/main/asciidoc/deploying-to-openshift.adoc:140]:
```suggestion
The base/builder image is specified using `base-jvm-image` and `base-native-image` for jvm and native mode respectively. An ImageStream for the image is automatically generated, unless these properties are used to reference an existing ImageStreamTag in the internal openshift registry. For example:
```

**Review by Sgitario — COMMENTED** [review]
LGTM. Only one comment about the name of the properties in the docs.

**Comment by iocanel** [comment]
@Sgitario: is there anything else?

**Review by Sgitario — APPROVED** [review]
lgtm now. You should squash the commits and fix the build.

**Pull request merged**